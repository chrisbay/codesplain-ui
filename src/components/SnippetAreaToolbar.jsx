import React, { PropTypes } from 'react';
import {
  TextField,
  Avatar,
} from 'material-ui';

import LanguageSelector from './LanguageSelector';
import LockButton from './buttons/LockButton';
import SaveMenu from './menus/SaveMenu';
import CustomPropTypes from '../util/custom-prop-types';

const styles = {
  toolbar: {
    backgroundColor: 'transparent',
    display: 'flex',
    flex: '1 0 auto',
    flexFlow: 'column wrap',
    margin: '0',
    padding: '0',
  },
  buttons: {
    flex: '1 1 auto',
  },
  toolbarField: {
    flex: '1 1 auto',
  },
  bottomRow: {
    display: 'flex',
    flexFlow: 'row wrap',
  },
  avatar: {
    flexBasis: 'auto',
    flexGrow: '1',
    flexShrink: '1',
    verticalAlign: 'middle',
  },
};

/*
https://www.w3schools.com/Css/css_navbar.asp
Note that the Material UI <Toolbar /> component does not work
here, because it does not reflow correctly when the screen is resized.
*/
const SnippetAreaToolbar = (props) => {
  const {
    canSave,
    language,
    onLanguageChange,
    onLockClick,
    onSaveAsClick,
    onSaveClick,
    onTitleChange,
    onOrgChanged,
    readOnly,
    saveEnabled,
    title,
    orgs,
    selectedOrg,
    avatarUrl,
  } = props;

  return (
    <div style={styles.toolbar}>
      <div style={styles.avatar}>
        {(avatarUrl) ?
          <Avatar
            size={30}
            src={avatarUrl}
          />
        : null}
      </div>
      <TextField
        hintText="Snippet Name"
        id="titleField"
        fullWidth
        onChange={onTitleChange}
        value={title}
      />
      <div style={styles.bottomRow} >
        <LanguageSelector
          disabled={readOnly}
          language={language}
          onChange={onLanguageChange}
          style={styles.toolbarField}
        />
        <div>
          <LockButton
            onClick={onLockClick}
            readOnly={readOnly}
            style={styles.buttons}
          />
          <SaveMenu
            canSave={canSave}
            enabled={saveEnabled}
            id="saveMenu"
            onSaveAsClick={onSaveAsClick}
            onSaveClick={onSaveClick}
            style={styles.buttons}
            orgs={orgs}
            onOrgChanged={onOrgChanged}
            selectedOrg={selectedOrg}
          />
        </div>
      </div>
    </div>
  );
};

SnippetAreaToolbar.propTypes = {
  canSave: PropTypes.bool.isRequired,
  language: PropTypes.string.isRequired,
  onLanguageChange: PropTypes.func.isRequired,
  onLockClick: PropTypes.func.isRequired,
  onSaveAsClick: PropTypes.func.isRequired,
  onSaveClick: PropTypes.func.isRequired,
  onTitleChange: PropTypes.func.isRequired,
  onOrgChanged: PropTypes.func.isRequired,
  readOnly: PropTypes.bool.isRequired,
  saveEnabled: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  orgs: CustomPropTypes.orgs.isRequired,
  selectedOrg: PropTypes.string,
};

SnippetAreaToolbar.defaultProps = {
  selectedOrg: '',
};

export default SnippetAreaToolbar;
