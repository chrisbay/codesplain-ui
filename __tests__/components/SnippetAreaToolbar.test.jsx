import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import SnippetAreaToolbar from '../../src/components/SnippetAreaToolbar';

describe('<SnippetAreaToolbar />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  it('matches snapshot', () => {
    const wrapper = shallowWithContext(
      <SnippetAreaToolbar 
        title={''}  
        onTitleChange={jest.fn()}
        readOnly={false}
        onLockClick={jest.fn()}
        onSaveClick={jest.fn()}
        onSaveAsClick={jest.fn()}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
