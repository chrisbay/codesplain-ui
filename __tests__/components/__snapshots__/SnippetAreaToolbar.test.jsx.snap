exports[`<SnippetAreaToolbar /> matches snapshot 1`] = `
<ul
  style={
    Object {
      "backgroundColor": "transparent",
      "listStyleType": "none",
      "margin": "0",
      "overflow": "hidden",
      "padding": "0",
    }
  }>
  <li
    style={
      Object {
        "display": "block",
        "float": "left",
        "padding": "5px 10px",
      }
    }>
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      fullWidth={false}
      hintText="Snippet Name"
      multiLine={false}
      onChange={[Function]}
      rows={1}
      style={
        Object {
          "verticalAlign": "middle",
          "width": "225px",
        }
      }
      type="text"
      underlineShow={true}
      value="" />
  </li>
  <li
    style={
      Object {
        "display": "block",
        "float": "left",
        "padding": "5px 10px",
      }
    }>
    <SelectField
      autoWidth={false}
      disabled={true}
      fullWidth={false}
      multiple={false}
      style={
        Object {
          "verticalAlign": "middle",
          "width": "120px",
        }
      }
      value={1}>
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        primaryText="Python 3"
        value={1} />
    </SelectField>
  </li>
  <li
    style={
      Object {
        "display": "block",
        "float": "left",
        "padding": "5px 10px",
      }
    }>
    <LockButton
      onClick={[Function]}
      readOnly={false}
      style={
        Object {
          "verticalAlign": "middle",
        }
      } />
    <SaveMenu
      onSaveAsClick={[Function]}
      onSaveClick={[Function]}
      style={
        Object {
          "verticalAlign": "middle",
        }
      } />
  </li>
</ul>
`;
